src/TestHelper/EnvironmentSpecificTestAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+[AttributeUsage(AttributeTargets.Class)]
+public class EnvironmentSpecificTestAttribute : Attribute, IApplyToContext
+{
+    readonly string[] environmentVariableNames;
+
+    public EnvironmentSpecificTestAttribute(params string[] environmentVariableNames) =>
+        this.environmentVariableNames = environmentVariableNames ?? Array.Empty<string>();
+
+    public void ApplyToContext(TestExecutionContext context)
+    {
+        var notPresentOrWhitespace = new List<string>();
+
+        foreach (var name in environmentVariableNames)
+        {
+            var value = Environment.GetEnvironmentVariable(name);
+
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                notPresentOrWhitespace.Add(name);
+            }
+        }
+
+        if (notPresentOrWhitespace.Count > 0)
+        {
+            Assert.Ignore($"Ignoring because environment variable(s) not present or white space: {string.Join(", ", notPresentOrWhitespace)}");
+        }
+
+        var presentAndNotWhitespace = new List<string>();
+
+        foreach (var name in EnvironmentVariables.Names.Except(environmentVariableNames))
+        {
+            var value = Environment.GetEnvironmentVariable(name);
+
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                presentAndNotWhitespace.Add(name);
+            }
+        }
+
+        Assert.Ignore($"Ignoring because environment variable(s) present and not white space: {string.Join(", ", presentAndNotWhitespace)}");
```

Comment:
I think I don't understand this section. This ignore is called regardless of everything else right?  It is not dependent on what is within presentAndNothWhitespace content.

Reply:
@WojcikMike this makes it so that tests marked with a single env var do not also run when both that env var and some other known env var are present.

For example, a test marked with only `AzureStorage_ConnectionString` will not run if both the `AzureStorage_ConnectionString`and `RabbitMQ_uri` env vars are present.

Otherwise, we will run the tests marked with `AzureStorage_ConnectionString` in all these jobs, when we only need them to run in the first job:

- AzureStorage
- AzureStorage-RabbitMQ
- AzureStorage-RavenDB4
- AzureStorage-SqlServer